packages/core/package.json

```diff
@@ -45,5 +47,8 @@
   "volta": {
     "extends": "../../package.json"
   },
+  "typesVersions": {
+    "<4.9": { "*": ["build/types-ts3.8/*"] }
```

Comment:
This creates a second folder in `build`, which is used by TS versions <4.9. We could tweak this, if we want, this would be the most straightforward config.